bitcoin/tests/test_rpc.py

```diff
@@ -13,24 +13,175 @@
 
 import unittest
 
-from bitcoin.rpc import Proxy
+import bitcoin.rpc
+from bitcoin.core import CBlock, CBlockHeader, lx, b2lx, COutPoint
+from bitcoin.core.script import CScript
+from bitcoin.wallet import CBitcoinAddress, CBitcoinSecret
+
+def is_active():
+    """
+    Proxy raises ValueError if cookie file not found
+    #FIXME is there a better way of doing this?
+    """
+    try:
+        p = bitcoin.rpc.Proxy()
+        p.help()
+        return True
+    except ValueError: 
+        return False
 
 class Test_RPC(unittest.TestCase):
+    _IS_ACTIVE = is_active()
     # Tests disabled, see discussion below.
     # "Looks like your unit tests won't work if Bitcoin Core isn't running;
     # maybe they in turn need to check that and disable the test if core isn't available?"
     # https://github.com/petertodd/python-bitcoinlib/pull/10
-    pass
-
-#    def test_can_validate(self):
-#        working_address = '1CB2fxLGAZEzgaY4pjr4ndeDWJiz3D3AT7'
-#        p = Proxy()
-#        r = p.validateAddress(working_address)
-#        self.assertEqual(r['address'], working_address)
-#        self.assertEqual(r['isvalid'], True)
-#
-#    def test_cannot_validate(self):
-#        non_working_address = 'LTatMHrYyHcxhxrY27AqFN53bT4TauR86h'
-#        p = Proxy()
-#        r = p.validateAddress(non_working_address)
-#        self.assertEqual(r['isvalid'], False)
+    # Sachin Meier: "I've changed it so each test checks against the _IS_ACTIVE variable "
+    #pass
+
+    def test_getbestblockhash_and_header(self):
+        if self._IS_ACTIVE:
+            proxy = bitcoin.rpc.Proxy()
+            blockhash = proxy.getbestblockhash()
+            header = proxy.getblockheader(blockhash)
+            self.assertTrue(isinstance(header, CBlockHeader))
+        else:
+            pass
+
+    def test_getblock(self):
+        if self._IS_ACTIVE:
+            proxy = bitcoin.rpc.Proxy()
+            blockhash = proxy.getbestblockhash()
+            # Test from bytes
+            block1 = proxy.getblock(blockhash)
+            self.assertTrue(isinstance(block1, CBlock))
+            # Test from str
+            block2 = proxy.getblock("0000000000000000000b4b0daf89eac9d84138fc900b8c473d4da70742e93dd0")
```

Comment:
This assumes that the daemon connected to by default tracks Bitcoin mainnet. This is totally not guaranteed.

Reply:
And in the automatic test environment spawning a full mainnet node will be too expensive.

Reply:
indeed. what are you proposing?

Reply:
special env var + regtest node, see my other comment, on test_getblockcount